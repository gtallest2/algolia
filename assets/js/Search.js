import React from 'react'

const { func } = React.PropTypes

const Search = React.createClass({
  propTypes: {
    onSearch: func,
    toggleMobile: func,
    clearSearch: func
  },

  getInitialState () {
    return {
      searchTerm: ''
    }
  },

  handleSearchTermChange (event) {
    this.props.onSearch(event.target.value)
    this.setState({ searchTerm: event.target.value })
  },

  openMobileMenu () {
    this.props.toggleMobile()
  },

  clearSearchQuery () {
    this.props.clearSearch()
    this.refs.searchInput.value = ''
    this.setState({ searchTerm: '' })
  },

  render () {
    return (
      <div className='search'>
        <div onClick={this.openMobileMenu} className='mobile-menu'><span className='filter-button'>Filter</span></div>
        <input ref='searchInput' type='text' onChange={this.handleSearchTermChange} placeholder='Search for Restaurants by Name, Cuisine, Location' />
        <div onClick={this.clearSearchQuery} className={(!this.state.searchTerm && 'invisible') + ' search-clear'}>X</div>
      </div>
    )
  }
})

export default Search
