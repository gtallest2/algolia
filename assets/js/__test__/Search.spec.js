import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Search from '../Search'

test('Search snapshot test', () => {
  const component = shallow(<Search />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
