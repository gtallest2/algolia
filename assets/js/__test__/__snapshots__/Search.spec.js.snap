exports[`test Search snapshot test 1`] = `
<div
  className="search">
  <div
    className="mobile-menu"
    onClick={[Function]}>
    <span
      className="filter-button">
      Filter
    </span>
  </div>
  <input
    onChange={[Function]}
    placeholder="Search for Restaurants by Name, Cuisine, Location"
    type="text" />
  <div
    className="invisible search-clear"
    onClick={[Function]}>
    X
  </div>
</div>
`;
